File: E-Olymp/0683_e_olymp.py
#17 saniye
m,n=map(int,input().split())
netice=[]
l = [list(map(int, input().split())) for _ in range(m)]
for i in range(m):
    for j in range(1,n):
        if i>0:
            l[i][0]=l[i-1][0]+l[i][0]
            break
        else:
            l[i][j]=l[i][j]+l[i][j-1]
for i in range(1,m):
    for j in range(1,n):
        l[i][j]=l[i-1][j]+l[i][j-1]-l[i-1][j-1]+l[i][j]
for setir in l:
    print(" ".join(map(str,setir)))




#easy wayyyy 14 san vaxt limiti asir
m, n = map(int, input().split())
l = [list(map(int, input().split())) for _ in range(m)]
s = [[0]*n for _ in range(m)]
for i in range(m):
    for j in range(n):
        s[i][j] = l[i][j]
        if i > 0:
            s[i][j] += s[i-1][j]
        if j > 0:
            s[i][j] += s[i][j-1]
        if i > 0 and j > 0:
            s[i][j] -= s[i-1][j-1]
for row in s:
    print(" ".join(map(str, row)))

File: E-Olymp/0683_e_olymp.cpp
//vaxt limiti asir
#include <stdio.h> 
#include <string.h>
using namespace std;
int main(void) {
    int i, j, n, m, value; 
    int mas[1000][1000]; 
    scanf("%d %d",&n,&m);
    for(i = 1; i <= n; i++)
    for(j = 1; j <= m; j++){
        scanf("%d",&value); 
        mas[i][j] = mas[i][j-1] + mas[i-1][j] - mas[i-1][j-1] + value;}
    for(i=1; i <= n; i++){
    for(j=1; j <= m; j++){
        printf("%d ",mas[i][j]);}
        printf("\n");}
    return 0; } 
